File: hw05/src/main/java/ru/otus/hw/models/Book.java
package ru.otus.hw.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.MappedCollection;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book {
    @Id
    private long id;

    private String title;

    private Author author;
    @MappedCollection(idColumn = "info_main_id")
    private List<Genre> genres;
}

File: hw10-front/src/app/model/Book.ts
import { BookModel } from './BookModel';
import { AuthorModel } from './AuthorModel';
import { GenreModel } from './GenreModel';

export class Book implements BookModel {

  public id: string;
  public title: string;
  public author: AuthorModel;
  public genres: GenreModel[];

  public constructor() {
    this.id = "";
    this.title = "";
    this.author = {} as AuthorModel;
    this.genres = [] as GenreModel[];
  }
}
